src/openrct2/world/Sprite.cpp

```diff
@@ -133,31 +133,54 @@ static void invalidate_sprite_max_zoom(SpriteBase* sprite, int32_t maxZoom)
     }
 }
 
-/**
- * Invalidate the sprite if at closest zoom.
- *  rct2: 0x006EC60B
- */
-void SpriteBase::Invalidate0()
+void SpriteBase::Invalidate()
 {
-    invalidate_sprite_max_zoom(this, 0);
-}
-
-/**
- * Invalidate sprite if at closest zoom or next zoom up from closest.
- *  rct2: 0x006EC53F
- */
-void SpriteBase::Invalidate1()
-{
-    invalidate_sprite_max_zoom(this, 1);
-}
-
-/**
- * Invalidate sprite if not at furthest zoom.
- *  rct2: 0x006EC473
- */
-void SpriteBase::Invalidate2()
-{
-    invalidate_sprite_max_zoom(this, 2);
+    int32_t maxZoom = 0;
+    switch (sprite_identifier)
+    {
+        case SpriteIdentifier::Vehicle:
+            maxZoom = 2;
+            break;
+        case SpriteIdentifier::Peep:
+            maxZoom = 0;
+            break;
+        case SpriteIdentifier::Misc:
+            switch (static_cast<MiscSpriteType>(type))
+            {
+                case MiscSpriteType::CrashedVehicleParticle:
+                    [[fallthrough]];
+                case MiscSpriteType::JumpingFountainWater:
+                    [[fallthrough]];
```

Comment:
my understanding of when you use [[fallthrough]] would be if you wanted to perform a bit of work and then fall through to the next case and not for just bunching together cases like this.

```
case 0:
   i++;
   [[fallthrough]]
case 1:
   std::cout << i;
   break;
```

Reply:
I remember a time where it complained, I can try it tho.